src/main.cpp

```diff
@@ -1258,6 +1258,18 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
         if (fRequireStandard && !AreInputsStandard(tx, view))
             return state.Invalid(false, REJECT_NONSTANDARD, "bad-txns-nonstandard-inputs");
 
+        // Check for bad witness
+        bool fNonStandardWitness = false;
+        if (!tx.wit.IsNull()) {
```

Comment:
I would do `if(witnessEnabled && !tx.wit.IsNull())`, as with it you can see what segwit is impacting by following where `witnessEnabled` is used. It makes review easier.